carbon-route-converter-behavior.html

```diff
@@ -0,0 +1,109 @@
+<script>
+  'use strict';
+
+  /**
+   * Provides bidirectional mapping between `path` and `queryParams` and a
+   * carbon-route compatible `route` object.
+   *
+   * For more information, see the docs for `carbon-route-converter`.
+   *
+   * @polymerBehavior
+   */
+  Polymer.CarbonRouteConverterBehavior = {
+    properties: {
+      /**
+       * A model representing the deserialized path through the route tree, as
+       * well as the current queryParams.
+       *
+       * A route object is the kernel of the routing system. It is intended to
+       * be fed into consuming elements such as `carbon-route`.
+       *
+       * @type {?Object}
+       */
+      route: {
+        type: Object,
+        notify: true
+      },
+
+      /**
+       * A set of key/value pairs that are universally accessible to branches of
+       * the route tree.
+       *
+       * @type {?Object}
+       */
+      queryParams: {
+        type: Object,
+        notify: true
+      },
+
+      /**
+       * The serialized path through the route tree. This corresponds to the
+       * `window.location.pathname` value, and will update to reflect changes
+       * to that value.
+       */
+      path: {
+        type: String,
+        notify: true,
+      }
+    },
+
+    observers: [
+      '_locationChanged(path, queryParams)',
+      '_routeChanged(route.prefix, route.path)',
+      '_routeQueryParamsChanged(route.__queryParams)'
+    ],
+
+    created: function() {
+      this.linkPaths('route.__queryParams', 'queryParams');
+      this.linkPaths('queryParams', 'route.__queryParams');
```

Comment:
Kevin corrected me on this: apparently `linkPaths` is bi-directional. Changing to only use one `linkPaths` in `<carbon-indexeddb-mirror>` seemed to prove this to be correct.